index.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
```

Comment:
`<?php
/**
 * qs_yurii_st.php
 *
 * @category
 * @author Vitaliy Pashkov<vpashkov@cartera.com>
 * @copyright   Copyright (c) 2021 - Cartera Commerce, Inc. All rights reserved.
 */


function bubleSort($arr, bool $order = true, $log = false, &$iterations = 0) {
    $iterations = 0;
    $lenght = count($arr);
//    echo 'Buble Sort, ' . ($order ? 'ASC' : 'DESC') . ' ';
//    echo ' - ';
//    echo 'Input array:';
//    print_r($arr);
    for($i = 0; $i < $lenght; $i++) {
        for($j = 0; $j < $lenght; $j++) {
            if(($arr[$i] > $arr[$j]) != $order) {
                $value = $arr[$j];
                $arr[$j] = $arr[$i];
                $arr[$i] = $value;
            }
            $iterations++;
            if($log) {
                echo $iterations . ': ';
                print_r($arr);
            }
        }
    }
//    echo 'Result in ' . $iterations . ' iterations : ' . PHP_EOL;
//    print_r($arr);

    return $arr;
}


function replace ( &$a, &$b ) {
    $c = $a;
    $a = $b;
    $b = $c;
}

function quickSortIteration (&$arr, &$iterations, $leftIndex, $rightIndex, $order, $log) {
    $pivotPoint = $arr[$rightIndex];
    $i = ($leftIndex - 1);

    for ($j = $leftIndex; $j <= $rightIndex- 1; $j++) {
        if (($arr[$j] > $pivotPoint) != $order) {
            $i++;
            replace ($arr[$i], $arr[$j]);
            $iterations++;
            if($log) {
                echo $iterations . ': ';
//                print_r($arr);
            }
        }
    }
    replace ($arr[$i + 1], $arr[$rightIndex]);
    $iterations++;
    if($log) {
        echo $iterations . ': ' .PHP_EOL;
//        print_r($arr);
    }

    return ($i + 1);
}

function quickSort ($arr, $order = true, $log = false, &$iterations = 0) {
//    echo 'Quick Sort, ' . ($order ? 'ASC' : 'DESC') . ' ';
//    echo ' - ';
//    echo 'Input array:';
//    print_r($arr);
    $iterations=0;
    $lenght = count($arr);
    $indexesStack = [0, $lenght-1];
    $top = 1;
    $iterations = 0;

    while ( $top >= 0 ) {
        $rightIndex = $indexesStack[ $top-- ];
        $leftIndex = $indexesStack[ $top-- ];

        $pivot = quickSortIteration( $arr, $iterations, $leftIndex, $rightIndex, $order, $log );

        if ( $pivot-1 > $leftIndex ) {
            $indexesStack[ ++$top ] = $leftIndex;
            $indexesStack[ ++$top ] = $pivot - 1;
        }

        if ( $pivot+1 < $rightIndex ) {
            $indexesStack[ ++$top ] = $pivot + 1;
            $indexesStack[ ++$top ] = $rightIndex;
        }
    }

//    echo 'Result in ' . $iterations . ' iterations : ';
//    print_r($arr);

    return $arr;
}

const SORT_ORDER_ASC  = 'ASC';
const SORT_ORDER_DESC = 'DESC';
/**
 * @param array $array
 * @return string
 */
function arrayToString(array $array)
{
    return implode(', ', $array);
}

/**
 * @param array $arr
 * @param string $order
 */
function tryBubble(array $arr, $order = SORT_ORDER_ASC)
{
    $result = bubleSort($arr, ($order === SORT_ORDER_ASC), false, $iterations);
    echo "Bubble sorted:" . arrayToString($result) . PHP_EOL;
    echo "          Iterations : " . $iterations . PHP_EOL;
}

/**
 * @param array $arr
 * @param string $order
 */
function tryQuickSort(array $arr, $order = SORT_ORDER_ASC)
{
    $result = quickSort($arr, ($order === SORT_ORDER_ASC),false, $iterations);
    echo "QuickSort sorted:" . arrayToString($result) . PHP_EOL;
    echo "          Iterations : " . $iterations . PHP_EOL;
}



$arr = [2, 7, 3, 1, 0, -10, 9, 5, 4, 19, 1, 0, 8, 8, 8];
echo str_repeat('-', 80) .PHP_EOL;
echo "Random input array:" .  arrayToString($arr). PHP_EOL . PHP_EOL;

tryBubble($arr);
tryQuickSort($arr);


echo str_repeat('-', 80) .PHP_EOL;
$arrSorted = [-10, 0, 0, 1, 1, 2, 3, 4, 5, 7, 8, 8, 8, 9, 19];
echo "Sorted input array:" .  arrayToString($arrSorted). PHP_EOL . PHP_EOL ;

tryBubble($arrSorted);
tryQuickSort($arrSorted);

echo str_repeat('-', 80) .PHP_EOL;
$arrWorst = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,];
echo "Equal value input array:" .  arrayToString($arrWorst). PHP_EOL . PHP_EOL ;

tryBubble($arrWorst);
tryQuickSort($arrWorst);

echo str_repeat('-', 80) .PHP_EOL;
$arrEmpty = [];
echo "Equal value input array:" .  arrayToString($arrEmpty). PHP_EOL . PHP_EOL ;

tryBubble($arrEmpty);
tryQuickSort($arrEmpty);

//
//$arrSorted = [-10, 0, 0, 1, 1, 2, 3, 4, 5, 7, 8, 8, 8, 9, 19];
//
//$bubleSortResult = bubleSort($arrSorted, false);
//$quickSortResult = quickSort($arrSorted, false);
//
//$arrSortedWorst = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1];
//
//$bubleSortResult = bubleSort($arrSortedWorst);
//$quickSortResult = quickSort($arrSortedWorst);
//
//$bubleSortResult = bubleSort([0,1,2,3,4,5,6,7,8,9,10]);
//$quickSortResult = quickSort([0,1,2,3,4,5,6,7,8,9,10]);
`